core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreAggregateCaseAggregations.java

```diff
@@ -410,13 +468,22 @@ private Optional<CaseAggregation> extractCaseAggregation(Symbol aggregationSymbo
                     toSqlType(aggregationType)));
         }
 
+        Expression whenOperand = caseExpression.getWhenClauses().getFirst().getOperand();
+        Expression whenResult = caseExpression.getWhenClauses().getFirst().getResult();
+        if (wrappedByCast ||
```

Comment:
Could we maybe extend it to simple aritmetic (e.g. mul, addition etc). @martint wdyt? Would it be better to whitelist here or blacklist?